crates/anvil/src/eth/otterscan/api.rs

```diff
@@ -183,51 +174,41 @@ impl EthApi {
 
         let best = self.backend.best_number();
         // we go from the first post-fork block, up to the tip
-        let from = if block_number == 0 {
-            self.get_fork().map(|f| f.block_number() + 1).unwrap_or(1)
-        } else {
-            block_number
-        };
+        let first_block = self.get_fork().map(|f| f.block_number() + 1).unwrap_or(1);
+        let from = if block_number == 0 { first_block } else { block_number + 1 };
         let to = best;
 
-        let first_page = from == best;
+        let mut first_page = from >= best;
         let mut last_page = false;
```

Comment:
nit: I believe you can use [`Itertools::with_position`](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.with_position) to not calculate `first`/`last` variables manually

Reply:
I suppose in both methods I can replace `n == to` with `pos == Position::Last` (lmk if you want this change and I will be happy to do it). However, the `first_page` and `last_page` booleans in the response refer to the entire list of transactions (does the response include the most recent transaction that ever touches this address? does it include the oldest one?), whereas the range of blocks in this loop might just be a subset of that complete range.

Reply:
Looking at this again, since we can't determine both `first_page` and `last_page` from the iterator, I think it would be clearer to keep the code as is.

Reply:
OK with this